### PR TITLE
[Android] Add option to disable video vpp if va driver is not ready.

### PR DESCRIPTION
**Pull request by js0701**
For certain reason on some platform the va driver is not enabled,
then video vpp path should be disabled.

Test: video playback normal and no other side effect.
Signed-off-by: Yuanjun Huang <yuanjun.huang@intel.com>
Singed-off-by: Lin Johnson <johnson.lin@intel.com>

### CODE REVIEW

**Comment by js0701** [comment]
This PR is not ready yet. Please don't use

**Comment by js0701** [comment]
@kalyankondapally @strassek @YuanjunHuang

**Comment by js0701** [comment]
@kalyankondapally Please review this patch suppose celadon is waiting for this

**Comment by csravelar** [comment]
@js0701 In order to pass travisci please run clang-format (version 3.9 or greater) -i --style=file on the following changed files :)

**Comment by js0701** [comment]
@csravelar Thanks. Will do so

File: os/platformcommondefines.h [hunk 1]
```diff
@@ -34,50 +34,8 @@ VkFormat NativeToVkFormat(int native_format);
 // minigbm specific DRM_FORMAT_YVU420_ANDROID enum
 #define DRM_FORMAT_YVU420_ANDROID fourcc_code('9', '9', '9', '7')
 
-/*
```
Comment (kalyankondapally) [inline os/platformcommondefines.h]: We have removed this patch in libdrm, master should be updated to use the latest libdrm

File: wsi/overlaybuffer.h [hunk 2]
```diff
@@ -71,12 +71,14 @@ class OverlayBuffer {
 
   virtual const ResourceHandle& GetGpuResource() = 0;
 
-  // Returns Media resource for this buffer which can be used by compositor.
-  // Surface will be clipped to width, height even if buffer size is
-  // greater than these values.
+// Returns Media resource for this buffer which can be used by compositor.
+// Surface will be clipped to width, height even if buffer size is
+// greater than these values.
+#ifndef DISABLE_VA
```
Comment (kalyankondapally) [inline wsi/overlaybuffer.h]: why comment this out ?

File: wsi/drm/drmbuffer.h [hunk 3]
```diff
@@ -79,9 +79,11 @@ class DrmBuffer : public OverlayBuffer {
 
   const ResourceHandle& GetGpuResource() override;
 
+#ifndef DISABLE_VA
   const MediaResourceHandle& GetMediaResource(MediaDisplay display,
```
Comment (kalyankondapally) [inline wsi/drm/drmbuffer.h]: why comment this out ?

File: wsi/drm/drmbuffer.cpp [hunk 4]
```diff
@@ -310,6 +316,7 @@ const ResourceHandle& DrmBuffer::GetGpuResource(GpuDisplay egl_display,
   return image_;
 }
 
+#ifndef DISABLE_VA
```
Comment (kalyankondapally) [inline wsi/drm/drmbuffer.cpp]: why comment this out ?

File: common/display/displayqueue.cpp [hunk 5]
```diff
@@ -349,7 +349,9 @@ void DisplayQueue::GetCachedLayers(const std::vector<OverlayLayer>& layers,
                    !target_plane.IsVideoPlane()) {
           const OverlayLayer& layer = layers.at(source_layers.at(0));
           if (layer.IsVideoLayer()) {
+#ifndef DISABLE_VA
             target_plane.SetVideoPlane(true);
```
Comment (kalyankondapally) [inline common/display/displayqueue.cpp]: Instead of commenting out all SetVideoPlane calls add this check in SetVideoPlane itself.

Comment (js0701) [reply common/display/displayqueue.cpp]: reasonable. Will do

File: wsi/overlaybuffer.h (hunk 2 above)
Comment (js0701) [reply wsi/overlaybuffer.h]: this function is only used if VA enabled. and It cause compiation issue in drmbuffer.cpp if va is not included. VA_SURFACE_INVALID is not defined. Anyway it make no sense to keep this function if VA is not enabled

File: wsi/drm/drmbuffer.h (hunk 3 above)
Comment (js0701) [reply wsi/drm/drmbuffer.h]: The same reason above

File: wsi/drm/drmbuffer.cpp (hunk 4 above)
Comment (js0701) [reply wsi/drm/drmbuffer.cpp]: Same reason obove

File: wsi/overlaybuffer.h (hunk 2 above)
Comment (kalyankondapally) [reply wsi/overlaybuffer.h]: Conceptually you could have different Media backend not using libva. If we are isolating libva changes, this is not right. We should have the implementation commented out with an error log saying "Function is not implemented" when libva is disabled. Also, enable this as one of the build step in travis ci.

Comment (js0701) [reply wsi/overlaybuffer.h]: Done

**Comment by kalyankondapally** [comment]
merged

**Pull request closed without merging**